reader/src/main/scala/za/co/absa/atum/reader/FlowReader.scala

```diff
@@ -18,21 +18,79 @@ package za.co.absa.atum.reader
+import za.co.absa.atum.reader.requests.QueryParamNames
 import za.co.absa.atum.reader.server.ServerConfig
 
 /**
  * This class is a reader that reads data tight to a flow.
- * @param mainFlowPartitioning  - the partitioning of the main flow; renamed from ancestor's 'flowPartitioning'
- * @param serverConfig          - the Atum server configuration
- * @param backend               - sttp backend, that will be executing the requests
- * @param ev                    - using evidence based approach to ensure that the type F is a MonadError instead of using context
- *                              bounds, as it make the imports easier to follow
- * @tparam F                    - the effect type (e.g. Future, IO, Task, etc.)
+ *
+ * @param mainFlowPartitioning - the partitioning of the main flow; renamed from ancestor's 'flowPartitioning'
+ * @param serverConfig         - the Atum server configuration
+ * @param backend              - sttp backend, that will be executing the requests
+ * @tparam F                   - the effect type (e.g. Future, IO, Task, etc.)
  */
-class FlowReader[F[_]](val mainFlowPartitioning: AtumPartitions)
-                      (implicit serverConfig: ServerConfig, backend: SttpBackend[F, Any], ev: MonadError[F])
-  extends Reader[F] with PartitioningIdProvider[F]{
+case class FlowReader[F[_]: MonadError](mainFlowPartitioning: AtumPartitions)
+                                       (implicit serverConfig: ServerConfig, backend: SttpBackend[F, Any])
+  extends Reader[F] with PartitioningIdProvider[F] {
+
+  /**
+   * Function to retrieve a page of checkpoints belonging to the flow.
+   * The checkpoints are ordered by their creation order.
+   *
+   * @param pageSize  - the size of the page (record count) to be returned
+   * @param offset    - offset of the page (starting position)
+   * @return          - a page of checkpoints
+   */
+  def getCheckpointsPage(pageSize: Int = 10, offset: Long = 0): F[RequestResult[PaginatedResponse[CheckpointWithPartitioningDTO]]] = {
+    for {
+      mainPartitioningIdOrError <- partitioningId(mainFlowPartitioning)
+      flowIdOrError <- mapRequestResultF(mainPartitioningIdOrError, queryFlowId)
+      checkpointsOrError <- mapRequestResultF(flowIdOrError, queryCheckpoints(_, None, pageSize, offset))
+    } yield checkpointsOrError
+  }
+
+  /**
+   * Function to retrieve a page of checkpoints of the given name belonging to the flow.
+   * The checkpoints are ordered by their creation order.
+   *
+   * @param checkpointName  - the name to filter with
+   * @param pageSize        - the size of the page (record count) to be returned
+   * @param offset          - offset of the page (starting position)
+   * @return                - a page of checkpoints
+   */
+  def getCheckpointsOfNamePage(checkpointName: String, pageSize: Int = 10, offset: Long = 0): F[RequestResult[PaginatedResponse[CheckpointWithPartitioningDTO]]] = {
```

Comment:
why not to have just 1 function with `checkpointName` being `Option[String]` ?

Reply:
I see it as inconvenient call.
And this way it clearly differentiate in the function name when only name-filtered entries are requested,
But can be unifies if seen as preferential.